packages/headless/src/features/commerce/context/cart/cart-slice.ts

```diff
@@ -0,0 +1,49 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {CartState, getCartInitialState} from './cart-state';
+import {addCartItem, removeCartItem, setCart, updateCartItemQuantity} from './cart-actions';
+import {ProductParam} from '../../../../api/commerce/commerce-api-params';
+
+const generateId = (product: ProductParam) => [
```

Comment:
Do we really need to do this?

The API expects the groupId, productId, and sku to be passed as separate arguments, so we'd have to convert them back anyways before making the request. If, for instance, groupId contained underscore characters before we joined all three parts, we'll have a hard time splitting it back as it must be.

Reply:
oh nm, I understand why you're doing this here hehe